src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -178,6 +182,143 @@ export abstract class ToggleableAction extends Action {
 	}
 }
 
+export class NotebookViewsOptions implements IActionProvider {
+	private _options: Action[];
+	private views: NotebookViewsExtension;
+	private viewMode: ViewMode;
+	private readonly _optionsUpdated = new Emitter<boolean>();
+
+	constructor(
+		container: HTMLElement,
+		views: NotebookViewsExtension,
+		modelReady: Promise<INotebookModel>,
+		@INotebookService private _notebookService: INotebookService,
+		@IInstantiationService private instantiationService: IInstantiationService) {
+
+		if (modelReady) {
+			modelReady
+				.then((model) => {
+					this.views = views;
+					this.viewMode = model.viewMode;
+					this.updateView();
+				})
+				.catch((err) => {
+					// No-op for now
+				});
+		}
+	}
+
+	getActions(): IAction[] {
+		return this._options;
+	}
+
+	public get options(): Action[] {
+		return this._options;
+	}
+
+	// Update SelectBox values
+	public updateView() {
+		const backToNotebookButton = this.instantiationService.createInstance(NotebookViewAction, 'notebookView.backToNotebook', localize('notebookViewLabel', 'Editor'), 'notebook-button');
+		const newViewButton = this.instantiationService.createInstance(CreateNotebookViewAction, 'notebookView.newView', localize('newViewLabel', 'Create New View'), 'notebook-button notebook-button-newview');
+
+		const views = this.views.getViews();
+		this._options = [];
+
+		this._options.push(backToNotebookButton);
+		this._options.push(newViewButton);
+
+		if (views.length) {
+			this._options.push(this.instantiationService.createInstance(Separator));
+		}
+
+		views.forEach((view) => {
+			const option = new DashboardViewAction(view.guid, view.name, 'button', this._notebookService);
+			this._options.push(option);
+
+			if (this.viewMode === ViewMode.Views && this.views.getActiveView() === view) {
+				option.checked = true;
+				option.enabled = false;
+			}
+		});
+
+		if (this.viewMode === ViewMode.Notebook) {
+			backToNotebookButton.checked = true;
+			backToNotebookButton.enabled = false;
+		}
+
+		this._optionsUpdated.fire(true);
+	}
+
+	public get onUpdated(): Event<boolean> {
+		return this._optionsUpdated.event;
+	}
+
+	public optionSelected(displayName: string): void {
+		const view = this.views.getViews().find(view => view.name === displayName);
+		this.views.setActiveView(view);
+	}
+}
+
+export class DashboardViewAction extends Action {
```

Comment:
It would be good to add doc comments for these classes explaining what the action is supposed to do.